lib/roku-develop.coffee

```diff
@@ -412,24 +412,26 @@ module.exports        = RokuDevelop =
     archive.pipe outputStream
 
     # Compile a list of files and directories to be compressed
-    for entry in @projectDirectory.getEntriesSync()
+    @addFilesToArchive(archive, @projectDirectory)
+
+    # Finish the compression, calling the output stream's close handler
+    archive.finalize()
+    return true
+
+  addFilesToArchive: (archive, parentEntry) ->
+    for entry in parentEntry.getEntriesSync()
       baseName = entry.getBaseName()
       pathname = entry.getRealPathSync()
       # Ignore hidden files and directories, and excluded files and directories
-      if  not (baseName.startsWith '.') and (baseName not in excludedPathList)
+      if  not (baseName.startsWith '.') and (baseName not in @excludedPathList)
         if entry.isFile()
           # Queue a file for compression, but not the zip file itself
           if pathname isnt @zipFilePath
-            archive.file pathname, {name: baseName}
+            archive.file pathname, {name: entry.getPath().replace(@projectDirectory.getPath(),"")}
         else if entry.isDirectory()
           # Queue a directory for compression, but not the zip file directory
           if pathname isnt path.dirname @zipFilePath
```

Comment:
Great work! The only change I would make so far is to handle the case where the zip file directory could be a symbolic link:

Change:
````
           if pathname isnt path.dirname @zipFilePath
````
to:
````
           zipDirectoryPath = path.dirname @zipFilePath
           if (pathname isnt zipDirectoryPath) and (entry.getPath() isnt zipDirectoryPath)
````

What do you think?

Reply:
Yes you're right. I'll update it